must.go

```diff
@@ -9,52 +9,74 @@
 //  }
 package must
 
-import "fmt"
+import (
+	"errors"
+	"fmt"
+)
+
+type wrappedError struct{ error }
+
+func (w wrappedError) Unwrap() error { return w.error }
 
 // Do returns a or panics if err != nil
 func Do[A any](a A, err error) A {
 	if err != nil {
-		panic(err)
+		panic(wrappedError{err})
 	}
 	return a
 }
 
 // Do0 panics if err != nil
 func Do0(err error) {
 	if err != nil {
-		panic(err)
+		panic(wrappedError{err})
 	}
 }
 
 // Do2 returns a and b or panics if err != nil
 func Do2[A, B any](a A, b B, err error) (A, B) {
 	if err != nil {
-		panic(err)
+		panic(wrappedError{err})
 	}
 	return a, b
 }
 
-// Handle sets err to recovered value if it is an error
-func Handle(err *error) {
-	v := recover()
-	if v == nil {
-		return
-	}
-	if err == nil {
-		panic(v)
-	}
-	if e, ok := v.(error); ok && e != nil {
-		*err = e
+// Handle sets pErr to recovered value if it is an error emitted inside of a Do call
+func Handle(pErr *error) {
+	e := recover()
+	handle(pErr, e)
+}
+
+func handle(pErr *error, e interface{}) {
+	switch {
```

Comment:
I cannot see the value of using "switch" instead of "if" here. Furthermore, the verification if "err == nil" was removed.

Reply:
valid point re:case statement, left cruft from some toying with the ideas. 

The verification is I believe unnessessary 'cause we will never get wrappedError{} from outside the Do call, where it's already checked for nil by definition. Either way if we decide to pass the nil error there, perhaps we know what we doing, like typed nil error values etc.